### PR TITLE
Add docstring for categorical function

### LINKED ISSUE #10: Exploratory Plots for categorical features
Exploratory Plots for categorical features
(input: x, y, color/fill, title, size of text, color; output: altair/matplot objects (e.g. histogram or bar charts) -

### PR DESCRIPTION
**Pull request by nichowu**

### CODE REVIEW

File: simpler_eda/categorical_eda.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    Parameters
+    ----------
+    data : pandas.core.frame.DataFrame
+       Input dataframe object.
+    xval : str
+      Variable used to represent the x-axis.
+    yval : str
+      Variable used to represent the y-axis.
+    plot_type : str, optional
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py:36]: specify options: "histogram", "density".
Also mention: if you pick density, y argument will be obsolete..

File: simpler_eda/categorical_eda.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import altair as alt
+import numpy as np
+import pandas as pd
+
+def categorical_plot(
+    data,
+    xval=None,
+    yval=None,
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py]: yval="count"

File: simpler_eda/categorical_eda.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    plot_type="bar",
+    color=None,
+    title=None,
+    font_size=10,
+    bar_size=5,
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py]: remove bar_size

File: simpler_eda/categorical_eda.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    color_scheme=None,
+    sort_by=None,
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py]: remove sort_by

File: simpler_eda/categorical_eda.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      Variable used to set the color of the marks in the plot object.
+    tilte : str, optional
+      Variable used to set the title of the plot.
+    font_size  : int, optional
+      Variable used to set the size of the axis labels and title.
+    bar_size : int, optional
+      Variable used to set the bar size.
+    color_scheme : str, optional
+      Variable used to set the bar size.
+    sort_by : str, optional
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py]: remove sort_by (not needed)

File: simpler_eda/categorical_eda.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    plot_type="bar",
+    color=None,
+    title=None,
+    font_size=10,
+    bar_size=5,
+    color_scheme=None,
+    sort_by=None,
+    plot_height=None,
+    plot_width=None,
+    facet=None
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py]: add:   
opacity=0.6

File: simpler_eda/categorical_eda.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    default arguements for color, plot title, size of text,
+    color-schem, orientation, plot size and facet factor.
+
+    Parameters
+    ----------
+    data : pandas.core.frame.DataFrame
+       Input dataframe object.
+    xval : str
+      Variable used to represent the x-axis.
+    yval : str
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py]: yval : str, optional

File: simpler_eda/categorical_eda.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    font_size=10,
+    bar_size=5,
+    color_scheme=None,
+    sort_by=None,
+    plot_height=None,
+    plot_width=None,
+    facet=None
+):
+    """
+    This function takes in a data frame object, one categorical column,
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py]: This function takes in a data frame object, one categorical feature to produce a histogram plot that visualizes the 
    distribution of the feature. User can also choose to plot density graph of the categorical feature. User can 
    optionally change the default arguments for colour, plot title, size of text,
    colour-scheme, orientation, plot size and facet factor.

File: simpler_eda/categorical_eda.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+      Variable used to represent the y-axis.
+    plot_type : str, optional
+      Variable used to represent the graphical relationship between xval and yval.
+    color : str, optional
+      Variable used to set the color of the marks in the plot object.
+    tilte : str, optional
+      Variable used to set the title of the plot.
+    font_size  : int, optional
+      Variable used to set the size of the axis labels and title.
+    bar_size : int, optional
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py]: bar_size - remove

File: simpler_eda/categorical_eda.py [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+      Variable used to set the bar size.
+    color_scheme : str, optional
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py:45]: color!!!

File: simpler_eda/categorical_eda.py [hunk 11]
```diff
@@ -0,0 +1,75 @@
+import altair as alt
+import numpy as np
+import pandas as pd
+
+def categorical_plot(
+    data,
+    xval=None,
+    yval=None,
+    plot_type="bar",
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py]: plot_type="histogram",

File: simpler_eda/categorical_eda.py [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+      Variable used to specify plot height
+    plot_witdh : int, optional
+      Variable used to specify plot width
+    facet : str, optional
+      Variable used to specify facet factor
+
+    Returns
+    -------
+    `altair`
+        A bar chart according to user specification in the parameters.
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py]: A histogram or density chart object based on user specification.

File: simpler_eda/categorical_eda.py [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+
+    Examples
+    --------
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py:62]: add: 
import altair as alt
import numpy as np
import pandas as pd

File: simpler_eda/categorical_eda.py [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+    >>> from simpler_eda.categorical_eda import categorical_plot
+    >>> from vega_datasets import data
+    >>> cars = data.cars()
+    >>> categorical_plot(cars, xval = "Origin", 
+                        yval = "count()",
+                        color = "Horsepower",
+                        title = "Histogram of Origin in Different Levels of Horsepower",
+                        sort_by = '-y',
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py]: remove:  sort_by

File: simpler_eda/categorical_eda.py [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+                        plot_height = 100,
+                        plot_width = 200
+                        )
+    """
```
Comment (nichowu) [inline simpler_eda/categorical_eda.py]: add:

return categorical_plot

File: simpler_eda/categorical_eda.py [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+    ----------
+    data : pandas.core.frame.DataFrame
+       Input dataframe object.
+    xval : str
+      Variable used to represent the x-axis.
+    yval : str, optional
+      Variable used to represent the y-axis.
+    plot_type : str, optional
+      Variable used to specify plot type. Options include "histogram" 
+      and "density". When "density" is selected, argument yval becomes obsolete.
```
Comment (ChuckHo777) [inline simpler_eda/categorical_eda.py]:
```suggestion
      and "density". When "density" is selected, the variable yval becomes obsolete.
```

File: simpler_eda/categorical_eda.py [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+    """
+    This function takes in a data frame object and one categorical
+    feature, to produce a histogram plot that visualizes the
+    distribution of the feature. User can also choose to plot density
+    graph of the feature by specifing in plot_type. 
+    The function also offers customization on color, plot title, 
+    font size, color-scheme, plot size, opacity level and facet 
+    factor.
+
+    Parameters
```
Comment (ChuckHo777) [inline simpler_eda/categorical_eda.py:28]: Do we need all the variables used to specify or represent in parameter/argument description?

File: simpler_eda/categorical_eda.py [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+
+def categorical_plot(
+    data,
+    xval = None,
+    yval = "count",
+    plot_type = "histogram",
+    color = None,
+    title = None,
+    font_size = 10,
+    color_scheme = "yellowgreenblue",
```
Comment (ChuckHo777) [inline simpler_eda/categorical_eda.py]: Should consider tableau20 color scheme instead of the sequential one as we are plotting for different categories.

**Review by ChuckHo777 — CHANGES_REQUESTED** [review]
added comments on default value on color scheme argument and parameter section of docstring.

File: simpler_eda/categorical_eda.py (hunk 18 above)
Comment (nichowu) [reply simpler_eda/categorical_eda.py]: Fixed in latest PR

File: simpler_eda/categorical_eda.py (hunk 17 above)
Comment (nichowu) [reply simpler_eda/categorical_eda.py:28]: Sorry I'm not really sure what you mean from above. Could you clarify? @ChuckHo777

Comment (nichowu) [reply simpler_eda/categorical_eda.py:28]: Good idea! I have updated the description.

**Review by ChuckHo777 — APPROVED** [review]
Great!

**Pull request merged**